music_assistant/server/providers/plex/__init__.py

```diff
@@ -0,0 +1,467 @@
+
+    @alru_cache(maxsize=128, ttl=300)
+    async def _get_data_cached(self, key, ma_cls):
+        if ma_cls == Artist:
+            return await self._parse_artist(await self._get_data(key, PlexArtist))
+        if ma_cls == Album:
+            return await self._parse_album(await self._get_data(key, PlexAlbum))
+
+    async def _parse(self, plex_media):
+        if plex_media.type == "artist":
+            return await self._parse_artist(plex_media)
+        elif plex_media.type == "album":
+            return await self._parse_album(plex_media)
+        elif plex_media.type == "track":
+            return await self._parse_track(plex_media)
+        elif plex_media.type == "playlist":
+            return await self._parse_playlist(plex_media)
+        return None
+
+    async def _search_track(self, search_query, limit) -> list[PlexTrack]:
+        return await self._run_async(
+            self._plex_library.searchTracks, title=search_query, limit=limit
+        )
+
+    async def _search_album(self, search_query, limit) -> list[PlexAlbum]:
+        return await self._run_async(
+            self._plex_library.searchAlbums, title=search_query, limit=limit
+        )
+
+    async def _search_artist(self, search_query, limit) -> list[PlexArtist]:
+        return await self._run_async(
+            self._plex_library.searchArtists, title=search_query, limit=limit
+        )
+
+    async def _search_playlist(self, search_query, limit) -> list[PlexPlaylist]:
+        return await self._run_async(self._plex_library.playlists, title=search_query, limit=limit)
+
+    async def _search_track_advanced(self, limit, **kwargs) -> list[PlexTrack]:
+        return await self._run_async(self._plex_library.searchTracks, filters=kwargs, limit=limit)
+
+    async def _search_album_advanced(self, limit, **kwargs) -> list[PlexAlbum]:
+        return await self._run_async(self._plex_library.searchAlbums, filters=kwargs, limit=limit)
+
+    async def _search_artist_advanced(self, limit, **kwargs) -> list[PlexArtist]:
+        return await self._run_async(self._plex_library.searchArtists, filters=kwargs, limit=limit)
+
+    async def _search_playlist_advanced(self, limit, **kwargs) -> list[PlexPlaylist]:
+        return await self._run_async(self._plex_library.playlists, filters=kwargs, limit=limit)
+
+    async def _search_and_parse(self, search_coro: Coroutine, parse_coro: Callable):
+        tasks = []
+        async with TaskGroup() as tg:
+            for item in await search_coro:
+                tasks.append(tg.create_task(parse_coro(item)))
+
+        results = []
+        for task in tasks:
+            results.append(task.result())
+
+        return results
+
+    async def _parse_album(self, plex_album: PlexAlbum) -> Album:
+        """Parse a Plex Album response to an Album model object."""
+        album_id = plex_album.key
+        album = Album(
+            item_id=album_id,
+            name=plex_album.title,
```

Comment:
can you swap the name and provider so item_id and provider are always the first entries ?